src/components/BookList.js

```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Book from "./Book";
+
+class BookList extends React.Component {
+  static propTypes = {
+    books: PropTypes.array.isRequired,
+    filters: PropTypes.object.isRequired
+  };
+
+  applyFilters = books => {
+    const filters = this.props.filters;
+    for (let filter in filters) {
+      if (filters[filter]) {
+        books = books.filter(book => book[filter]);
```

Comment:
This looks a bit clunky — it works because my filters are booleans, so `if(filters[filter])` checks if the filter is set to `true`, and if so, then it filters the book array by `book[filter]` which checks if the book matches. e.g.:

```
if (filters[purchased]) books = books.filter(book => book[purchased]);
```
Hmm. Maybe if I feel the need to explain it here, I should have a comment in the code.